### ISSUE TITLE
Interactive: command_not_found_handle

### ISSUE THREAD

**Issue by Melkor333** [oilshell/oil#1845]
When a Command is not found, bash executes a `command_not_found_handle` ([manpage](https://www.gnu.org/software/bash/manual/html_node/Command-Search-and-Execution.html) ).

I don't think `osh` should necessarily implement this, but I think it makes sense to at least gather the missing interactive features.

**Comment by andychu** [comment]
Yeah I expect someone to ask for this at some point :)  I haven't run into it myself

**Comment by Melkor333** [comment]
That one is me :p Nix has a convenient function showing a possible `nix-shell -p PACKAGE` command when a package isn't found. :)
Do you think it makes sense to implement this the same way bash does or another way? (or not at all - people could just use FANOS and a custom frontend?)

**Comment by andychu** [comment]
Oh yes I think we can do this exactly like bash

If you run into an actual usage, don't be shy about requesting it

We prefer the bugs that have a real use case, as opposed to "This is in the bash manual (or not in the manual) and we don't know if anyone uses it)
